charts/consul/templates/crd-gatewaypolicies.yaml

```diff
@@ -0,0 +1,208 @@
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: GatewayPolicySpec defines the desired state of GatewayPolicy.
+            properties:
+              default:
+                properties:
+                  jwt:
+                    description: APIGatewayJWTRequirement holds the list of JWT providers
+                      to be verified against.
+                    properties:
+                      providers:
+                        description: Providers is a list of providers to consider
+                          when verifying a JWT.
+                        items:
+                          description: APIGatewayJWTProvider holds the provider and
+                            claim verification information.
+                          properties:
+                            name:
+                              description: Name is the name of the JWT provider. There
+                                MUST be a corresponding "jwt-provider" config entry
+                                with this name.
+                              type: string
+                            verifyClaims:
+                              description: VerifyClaims is a list of additional claims
+                                to verify in a JWT's payload.
+                              items:
+                                description: APIGatewayJWTClaimVerification holds
+                                  the actual claim information to be verified.
+                                properties:
+                                  path:
+                                    description: Path is the path to the claim in
+                                      the token JSON.
+                                    items:
+                                      type: string
+                                    type: array
+                                  value:
+                                    description: "Value is the expected value at the
+                                      given path: - If the type at the path is a list
+                                      then we verify that this value is contained
+                                      in the list. \n - If the type at the path is
+                                      a string then we verify that this value matches."
+                                    type: string
+                                required:
+                                - path
+                                - value
+                                type: object
+                              type: array
+                          required:
+                          - name
+                          type: object
+                        type: array
+                    required:
+                    - providers
+                    type: object
+                required:
+                - jwt
```

Comment:
Is the `omitEmpty` json tag what adds/removes things from the `required` list in the resulting CRD?

Reply:
I believe `kubebuilder` defaults everything to `required`, but you can explicitly mark fields with a `+optional` comment if you want them to be optional in the JSON spec. I believe if the generator sees a `json` tag with `omitempty` it also interprets the field as having a `+optional` comment added to it.

Reply:
this is now optional